source/documentation/deploying-an-app/ingress-with-duplicate-hostnames.html.md.erb

```diff
@@ -0,0 +1,135 @@
+
+## Required Annotations
+To allow an Ingress to use a hostname that is already in use in another namespace, you **must** add the following three annotations to each Ingress resource (**both** the new and the existing ones):
+
+```
+# Indicates that you intentionally want to allow duplicate hostnames.
+allow-duplicate-host: "true"
+
+# A comma-separated list of namespaces where duplicate hostnames are permitted.
+# IMPORTANT: The value must be identical and in the smae order on both Ingress objects.
+allowed-duplicate-ns: "blue,green"
+
+# Specifies the AWS weight value used for traffic balancing.
+# This value must be non-empty.
+external-dns.alpha.kubernetes.io/aws-weight: "100"
+
+# A unique identifier for the Ingress.
+external-dns.alpha.kubernetes.io/set-identifier: "<ingress-name>-<ns>-${clusterColor}"
+```
+
+## Canary Deployment Example
+
+Because the NGINX Ingress Controller normally does not allow two Ingress objects to define the same host and path combination, you must enable canary ingresses to merge configurations. To do so, add the following canary annotations:
+
+```
+# Marks the Ingress as part of a canary (blue-green) deployment.
+nginx.ingress.kubernetes.io/canary: "true"
+
+# Defines the traffic weight for the canary.
+nginx.ingress.kubernetes.io/canary-weight: "50"
+```
+
+- Baseline Weight Note:
+
+Although both `external-dns.alpha.kubernetes.io/aws-weight` and `nginx.ingress.kubernetes.io/canary-weight` are available, it is often simpler to use a single weight annotation as your baseline for traffic splitting. Consolidating to one value reduces configuration complexity and potential confusion. 
+
+Depending on your environment and the controller’s behavior, you might choose to adjust `nginx.ingress.kubernetes.io/canary-weight` for Canary weighting deployment.
+
+
+## How It Works
+When you create or update an Ingress, Gatekeeper policies check for duplicate hostnames across namespaces. If the hostname is already in use in another namespace, then both the new Ingress and the existing Ingress must include all required annotations. In particular, the value of `allowed-duplicate-`ns must be identical and in same order on both Ingresses. If the values differ, the request will be rejected.
+
+Because the NGINX Ingress Controller normally prevents duplicate host/path combinations, using the canary annotations instructs it to merge the configuration of your Ingress objects. This enables you to use the same host and same path in a blue–green deployment.
+
+### Example: Using Canary Deployment with Duplicate Hostname Across Namespaces
+Below are sample YAML files for a Canary deployment scenario where duplicate hostnames across namespaces are allowed because the required annotations are present and identical.
+
+#### Inventory Ingress (Existing Deployment)
+This Ingress represents your blue deployment in the `blue` namespace. It is configured as the master Ingress (mergeable) and includes the required blue–green annotations.
```

Comment:
We are merging 2 different deployment strategies here. On the one hand there is blue green and on the other there is canary. I think confusingly we should refer to different namespaces under the canary deployment strategy as namespace a and namespace b